src/VideoCommand.cc

```diff
@@ -350,3 +373,202 @@ Json::Value FindVideo::construct_responses(
+    bool is_frames   = cmd.isMember("frames");
+
+    if (!(is_frames != is_interval)) {
+        error["status"]  = RSCommand::Error;
+        error["info"] = "Either one of 'frames' or 'operations::interval' "
+                        "must be specified";
+        return -1;
+    }
+
+    Json::Value results = get_value<Json::Value>(cmd, "results");
+    results["list"].append(VDMS_VID_PATH_PROP);
+
+    query.QueryNode(
+            get_value<int>(cmd, "_ref", -1),
+            VDMS_VID_TAG,
+            cmd["link"],
+            cmd["constraints"],
+            results,
+            get_value<bool>(cmd, "unique", false)
+            );
+
+    return 0;
+}
+
+Json::Value FindFrames::construct_responses(
+    Json::Value& responses,
+    const Json::Value& json,
+    protobufs::queryMessage &query_res,
+    const std::string &blob)
+{
+    const Json::Value& cmd = json[_cmd_name];
+
+    Json::Value ret;
+
+    auto error = [&](Json::Value& res)
+    {
+        ret[_cmd_name] = res;
+        return ret;
+    };
+
+    Json::Value resp = check_responses(responses);
+    if (resp["status"] != RSCommand::Success) {
+        return error(resp);
+    }
+
+    Json::Value& FindFrames = responses[0];
+
+    bool flag_empty = true;
+
+    for (auto& ent : FindFrames["entities"]) {
+
+        std::string video_path = ent[VDMS_VID_PATH_PROP].asString();
+        ent.removeMember(VDMS_VID_PATH_PROP);
+
+        if (ent.getMemberNames().size() > 0) {
+            flag_empty = false;
+        }
+
+        try {
+            std::vector<unsigned> frames;
+
+            // Copy of operations is needed, as we pass the operations to
+            // the enqueue_operations() method of ImageCommands class, and
+            // it should not include 'interval' operation.
+            Json::Value operations  = cmd["operations"];
+
+            Json::ArrayIndex interval_idx;
+            bool is_interval = get_interval_index(cmd, interval_idx);
+            bool is_frames   = cmd.isMember("frames");
+
+            if (is_frames) {
+                for (auto& fr : cmd["frames"]) {
+                    frames.push_back(fr.asUInt());
+                }
+            }
+            else if (is_interval) {
+
+                Json::Value interval_op = operations[interval_idx];
+
+                int start = get_value<int>(interval_op, "start");
+                int stop  = get_value<int>(interval_op, "stop");
+                int step  = get_value<int>(interval_op, "step");
+
+                for (int i = start; i < stop; i += step)  {
+                    frames.push_back(i);
+                }
+
+                Json::Value deleted;
+                operations.removeIndex(interval_idx, &deleted);
+            }
+            else {
+                // This should never happen, as we check this condition in
+                // FindFrames::construct_protobuf(). In case this happens, it
+                // is better to signal it rather than to continue
```

Comment:
If this cannot happen, shouldn't this be an assert since it sounds more like a code problem.

Reply:
It is better to not have assertions and doing it this way as it will log the problem rather than killing the server. Killing the server with an assertion is very painful when writing applications. It is more robust to return an error and log the problem, and we have the comment that explains what can be happening for debug. We need to work in the future on better logging strategies as well.

Reply:
Depends on whether you are trying to address a user created problem or problem with the server code. 
Agree with the logging comment.